src/core/utils/slaac_address.cpp

```diff
@@ -291,7 +290,7 @@ otError Slaac::GenerateIid(Ip6::NetifUnicastAddress &aAddress,
 
     GetIidSecretKey(secretKey);
 
-    for (uint16_t count = 0; count < kMaxIidCreationAttempts; count++, dadCounter++)
+    for (uint8_t count = 0; count < kMaxIidCreationAttempts; count++, dadCounter++)
```

Comment:
```suggestion
    for (int count = 0; count < kMaxIidCreationAttempts; count++, dadCounter++)
```

`count < kMaxIidCreationAttempts` is always true ?

Reply:
good catch 👍